File: z/从尾到头打印链表.cc
/**
 *  struct ListNode {
 *        int val;
 *        struct ListNode *next;
 *        ListNode(int x) :
 *              val(x), next(NULL) {
 *        }
 *  };
 */
class Solution {
 public:
  vector<int> printListFromTailToHead(ListNode* head) {
    vector<int> res;
    if (head == NULL) {
      return res;
    }
    while (head != NULL) {
      res.push_back(head->val);
      head = head->next;
    }
    reverse(begin(res), end(res));
    return res;
  }
};

File: h/从尾到头打印链表.java
/**
*    public class ListNode {
*        int val;
*        ListNode next = null;
*
*        ListNode(int val) {
*            this.val = val;
*        }
*    }
*
*/
import java.util.ArrayList;
import java.util.Collections;
public class Solution {
    public ArrayList<Integer> printListFromTailToHead(ListNode listNode) {
       
        ArrayList<Integer> list = new ArrayList<Integer>();
         
        if (listNode != null) {
           ListNode p = listNode;
  
            while (p != null) {
                list.add(p.val);
                p = p.next;
            }         
        }
        ArrayList<Integer> a = new ArrayList<>();
        Collections.reverse(list);
        return list;
    }
}
